spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/MailHealthIndicator.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2012-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.health;
+
+import java.net.URI;
+
+import javax.mail.MessagingException;
+
+import org.springframework.boot.actuate.health.Health.Builder;
+import org.springframework.mail.javamail.JavaMailSenderImpl;
+
+/**
+ * @author Johannes Stelzer
+ */
+public class MailHealthIndicator extends AbstractHealthIndicator {
+	private final JavaMailSenderImpl mailSender;
+
+	public MailHealthIndicator(JavaMailSenderImpl mailSender) {
+		this.mailSender = mailSender;
+	}
+
+	@Override
+	protected void doHealthCheck(Builder builder) throws Exception {
+		URI uri = new URI(this.mailSender.getProtocol(), null, this.mailSender.getHost(),
+				this.mailSender.getPort(), null, null, null);
+		builder.withDetail("connection", uri.toString());
+		try {
+			this.mailSender.getSession().getTransport().connect();
+			builder.up();
```

Comment:
Well, that's bad. The `Transport` instance is never closed. `JavaMailSenderImpl#connectTransport` looks like a good fit for what you're doing (relying on the plain `connect` does not seem such a great idea either). Unfortunately it's protected but not rocket science either.

I guess you need a `Transport#close()` class in a guarded finally block code.